lib/Runtime/Types/DictionaryTypeHandler.cpp

```diff
@@ -368,9 +368,9 @@ namespace Js
         DictionaryPropertyDescriptor<T> descriptor(index, attributes);
 #if ENABLE_FIXED_FIELDS
         Assert((!isFixed && !usedAsFixed) || (!IsInternalPropertyId(propertyRecord->GetPropertyId()) && this->singletonInstance != nullptr));
-        descriptor.IsInitialized = isInitialized;
-        descriptor.IsFixed = isFixed;
-        descriptor.UsedAsFixed = usedAsFixed;
+        descriptor.SetFlag(DictionaryPropertyDescriptor<T>::Flags::IsInitialized, isInitialized);
```

Comment:
nit: would be nice to have `typedef DictionaryPropertyDescriptor<T>::Flags Flags` or something to simplify notation